pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -164,12 +165,28 @@ func newAlertmanager(orgID int64, cfg *setting.Cfg, store store.AlertingStore, k
 	c := am.peer.AddState(fmt.Sprintf("notificationlog:%d", am.OrgID), am.notificationLog, m.Registerer)
 	am.notificationLog.SetBroadcast(c.Broadcast)
 
+	//TODO yuriy. replace with silencesFilePath when fix in https://github.com/prometheus/alertmanager/pull/2710 is released
+	silenceOpts := silence.Options{
+		Metrics:   m.Registerer,
+		Retention: retentionNotificationsAndSilences,
+	}
+
+	//The path is generated by the filestore. So presumably it should be safe
+	//nolint:gosec
+	silencesFileReader, err := os.Open(silencesFilePath)
```

Comment:
Given this is temporary, how do you feel about placing this code within the `fileStore`? `newAlertmanager` is already too big as it is and I'd love to keep stuff out unless it makes it better.

You can either create a new method instead of adding more to `FilepathFor` - the micro optimisation of using a buffer instead of reading from disk is not something that we necessarily _need_ given the restore from a file only happens on once (*per the lifetime of the org) and on start-up. Subsequent config changes get applied at runtime. 

But I do understand that the bug of closing the file descriptor should be fixed.

Reply:
I am not sure I can see how it can be improved by moving code to fileStore. I would appreciate your ideas. I think the simplest possible solution would be to just move intiailization to a different function/

Reply:
Nothing to worry about - the `fileStore` was a suggestion because you've left a comment referring to something that the fileStore already does so it seems like a dependency - this is good enough for now. good work.